arduino-ide-extension/src/browser/widgets/sketchbook/sketchbook-widget-contribution.ts

```diff
@@ -230,4 +261,31 @@ export class SketchbookWidgetContribution
   protected onCurrentWidgetChangedHandler(): void {
     this.selectWidgetFileNode(this.shell.currentWidget);
   }
+
+  protected async showSketchbookWidget(): Promise<void> {
+    this.widget
+      .then((widget) => this.shell.activateWidget(widget.id))
+      .then((widget) => {
+        if (widget instanceof SketchbookWidget) {
+          widget.activateTreeWidget(widget.getTreeWidget().id);
+          if (this.editorManager.currentEditor)
+            this.shell.activateWidget(this.editorManager.currentEditor.id);
+        }
+      });
+  }
+
+  protected runEncodedCommands(): void {
+    const params = new URLSearchParams(window.location.search);
+    const encoded = params.get('commands');
+    if (!encoded) return;
+
+    const commands = JSON.parse(decodeURIComponent(encoded));
+
+    if (commands && Array.isArray(commands)) {
+      commands.forEach((c: Command) => {
+        if (this.commandRegistry.commandIds.includes(c.id))
```

Comment:
This does not make much sense. No need to be fault-tolarent.

I prefer seeing an error, to silently ignoring a command. Please change.